### PR TITLE
Nav bar

### PR DESCRIPTION
**Pull request by JasonPKnoll**
## Implemented features:
- Refactored application.html.erb and added header partial
- Added routing functionality to navbar
- Added helper_method current_user in application controller
- Add destroy route for sessions
- Add destroy method in sessions controller

## Roadblocks/Logs/Errors:
X

## Testing:
- Add test for route redirects in the navbar
   - Dashboard
   - Home
   - Login
   - Logout

## Notes/For the future:
X

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -42,7 +43,7 @@
           </ul>
         </div>
       </div>
-    </nav>
+    </nav> -->
     <div class="container">
       <div class="row justify-content-center">
```
Comment (AlexKlick) [inline app/views/layouts/application.html.erb:49]: for this flash message, we can use       

`
        <% flash.each do |type, msg| %>
          <div align="center">
            <div class="alert alert-<%=type%>">
              <%= msg  %>
            </div>
          </div>
        <% end %>
  `

and then use the [bootstrap alert classes](https://getbootstrap.com/docs/5.0/components/alerts/) to give colors based on the result of the flash

**Review by AlexKlick — APPROVED** [review]
really solid, good work!

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -12,7 +12,8 @@
   </head>
 
   <body>
-    <nav class="navbar navbar-expand-lg navbar-dark bg-dark">
+    <%= render partial: 'welcome/header', locals: { placeholder: ''} %>
```
Comment (marlitas) [inline app/views/layouts/application.html.erb:15]: Great idea to put this out into a partial

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -27,10 +12,15 @@
...
     .to_return(status: 200, body: WebmockStubs.mock_artwork_recommendations, headers: {})
-    # stub_request(:get, 'https://www.localhost:3000/api/v1/art/4')
-    # .to_return(status: 200, body: WebmockStubs.mock_artwork_recommendations, headers: {})
+    stub_request(:get, 'https://www.localhost:3000/api/v1/art/4')
+    .to_return(status: 200, body: WebmockStubs.mock_artwork_recommendations, headers: {})
+    stub_request(:get, 'https://www.localhost:3000/api/v1/users/1/rated_art/4')
+    .to_return(status: 200, body: WebmockStubs.mock_artwork_recommendations, headers: {})
     stub_request(:get, 'https://www.localhost:3000/api/v1/art/1')
     .to_return(status: 200, body: WebmockStubs.mock_artwork_recommendations, headers: {})
+    stub_request(:get, 'https://www.localhost:3000/api/v1/users/1/rated_art/1')
```
Comment (marlitas) [inline spec/spec_helper.rb]: ooh there might be a conflict here we changed all the urls to connect to our backend peaceful-reef heroku. Definitely something to keep an eye out for in merge.

**Review by marlitas — APPROVED** [review]
Just need to keep an eye out for the stub links on the merge.

**Pull request merged**